azurerm/resource_arm_api_management_openid_connect_provider.go

```diff
@@ -0,0 +1,170 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2018-01-01/apimanagement"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmApiManagementOpenIDConnectProvider() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApiManagementOpenIDConnectProviderCreateUpdate,
+		Read:   resourceArmApiManagementOpenIDConnectProviderRead,
+		Update: resourceArmApiManagementOpenIDConnectProviderCreateUpdate,
+		Delete: resourceArmApiManagementOpenIDConnectProviderDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": azure.SchemaApiManagementChildName(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"api_management_name": azure.SchemaApiManagementName(),
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
```

Comment:
Should we validate this is an id rather then just not empty?

Reply:
this is a user-specifyable Client ID, so could be anything unfortunately